pkg/provision/local_store.go

```diff
@@ -102,13 +105,34 @@ func NewFSStore(root string) (*FSStore, error) {
 
 	log.Info().Msg("restart detected, keep reservation cache intact")
 
-	store := &FSStore{
-		root: root,
-	}
-
 	return store, store.sync()
 }
 
+func (s *FSStore) removeAllButPersistent(rootPath string) error {
+	err := filepath.Walk(rootPath, func(path string, info os.FileInfo, r error) error {
+		if r != nil {
+			return r
+		}
+		if info.IsDir() {
+			return nil
+		}
+		reservation, err := s.Get(filepath.Base(path))
```

Comment:
Since all you care from the reservation serialized on disk is the type, you can have an inline type that you use inside this function.
Something like
```go
res := struct{
	Type ReservationType `json:"type"`
}{}
```
So the unmarshalling of the data goes faster.